v2/spanner-custom-shard/src/main/java/com/custom/CustomShardIdFetcherForIT.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (C) 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.custom;
+
+import com.google.cloud.teleport.v2.spanner.utils.IShardIdFetcher;
+import com.google.cloud.teleport.v2.spanner.utils.ShardIdRequest;
+import com.google.cloud.teleport.v2.spanner.utils.ShardIdResponse;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class CustomShardIdFetcherForIT implements IShardIdFetcher {
```

Comment:
It cannot be moved inside, it needs to be inside a java package so that JAR generation happens correctly, will add unit test to cover these, if that's okay?